### ISSUE TITLE
Install XCode / clang 12.4 to support Java 17+

### ISSUE THREAD

**Issue by jdekonin** [eclipse-openj9/openj9#15169]
As per https://wiki.openjdk.java.net/display/Build/Supported+Build+Platforms 11.2 with XCode / clang 12.4 is the supported compiler for Java 17+. OpenJ9 is currently documented as using 10.14 with xcode 10.3 and clang 10.0.1 for all versions.

I have 12.4 working with my internal systems after https://github.com/eclipse/omr/pull/5800 was merged.

As discovered working though this on my internal system, the minimum platform for jdk17+ should be 10.15 which also happens to be the minimum level xcode v12.4 will work on.  This means that OpenJ9 system(s) need to upgraded first.  Side note JDK8 & JDK11 will also build on 12.4 so we could keep the setup/configuration minimal by keeping the same version of xcode across all releases.

**Comment by jdekonin** [comment]
Listing of Macs
https://openj9-jenkins.osuosl.org/label/sw.os.mac&&hw.arch.x86/

10.11.6
 - [ ] osx1011-x86-1
 - [ ] osx1011-x86-2
10.13.6
 - [ ] osx1013-x86-3
 - [ ] osx1013-x86-4
10.14.6
 - [ ] osx1014-x86-1
 - [ ] osx1014-x86-2
 - [ ] osx1014-x86-3
10.15.7
 - [ ] mac10-x86-1
 - [ ] mac10-x86-2

@pshipton what do you think would be the ideal version coverage?  I imagine we should likely also have 1 or more v11.X and v12.X systems?

**Comment by pshipton** [comment]
>  Side note JDK8 & JDK11 will also build on 12.4 so we could keep the setup/configuration minimal by keeping the same version of xcode across all releases.

Yes, this is fine with me.

**Comment by pshipton** [comment]
> I think we'd be fine with one 10.11

Actually ATM we aren't using either of these machines as they both have problems, either due to the hardware or the OS version.

**Comment by pshipton** [comment]
What we could do with 10.11 is remove the test label so it's not used for regular testing, but then setup a minimal test that runs nightly to ensure we can still startup on this OS version.

**Comment by jdekonin** [comment]
> What do you mean by that, the minimum platform for building or running? We have the runtime documented as 10.14 for jdk17, and 10.15 for jdk18+. I think we can build jdk17 on 10.15 and still run on 10.14.

I actually took this part from 1 of your comments.  I guess I misunderstood.  No worries I wasn't planning on touching labels used for targeting testing.  But from that comment I am guessing you'll want to keep a 10.14 system.  I've updated the description above with upgrade plan.  Thoughts?

**Comment by pshipton** [comment]
I don't see the upgrade plan.

**Comment by jdekonin** [comment]
Sorry, I guess I didn't save the comment.  Its there now as an update to the second comment in this issue; _**bold and italics.**_

**Comment by pshipton** [comment]
It would be better to have two 12.x machines rather than 3 11.x
I'm not sure if osx1011-x86-2 will be a usable machine, is it upgradable to 12.x?
We've this persistent problem https://github.com/eclipse-openj9/openj9/issues/9782 on osx1011-x86-2, it will be interesting to see if an OS upgrade makes it stable.

**Comment by jdekonin** [comment]
I updated to ave 2 * 11 and 2 * 12.  Not sure we can upgrade either 10.11 machine due to age of the hardware, but I'll try.  I could also leave osx1011-x86-2 as is and upgrade osx1011-x86-1 to avoid the problematic hardware, but that would remove the option for stable targeted testing.

**Comment by pshipton** [comment]
>  I could also leave osx1011-x86-2 as is and upgrade osx1011-x86-1 to avoid the problematic hardware, but that would remove the option for stable targeted testing.

Yes, let's do that. We don't need to do a lot of testing, basically just ensure the JVM runs. We can try to avoid running tests that would fail.

**Comment by jdekonin** [comment]
osx1011-x86-1 didn't seem to like to upgrade.  I attempted the same actions on osx1011-x86-2 and before I knew it I ad lost connection and it was happily upgrading. :(

I relaunched a nightly with targeted testing at mac11-x86-1 (which use to be osx1011-x86-2).  There is no `ci.role.test` label atm, so no worries about jobs landing on this system yet.
https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK11_aarch64_linux/747/

**Comment by jdekonin** [comment]
I've added `ci.role.test` to mac11-x86-1 since it passed all testing.  I found a previous issue open (#7050) that suggested system.special hangs and while this isn't in the nightly testing atm, I spun one up to see what happens after the upgrade.

https://openj9-jenkins.osuosl.org/job/Grinder/896/

Working on osx1013-x86-3 next

**Comment by pshipton** [comment]
There was also the following on [mac11-x86-1](https://openj9-jenkins.osuosl.org/computer/mac11-x86-1)
```
02:28:56  ...
02:28:56  ... TestNG 6.14.2 by Cédric Beust (cedric@beust.com)
02:28:56  ...
02:28:56  
02:29:17  Cannot contact mac11-x86-1: java.lang.InterruptedException
02:38:51  wrapper script does not seem to be touching the log file in /Users/jenkins/workspace/Test_openjdk18_j9_extended.functional_x86-64_mac_Nightly_testList_0@tmp/durable-cdf76a05
02:38:51  (JENKINS-48300: if on an extremely laggy filesystem, consider -Dorg.jenkinsci.plugins.durabletask.BourneShellScript.HEARTBEAT_CHECK_INTERVAL=86400)
```
https://openj9-jenkins.osuosl.org/job/Test_openjdk18_j9_extended.functional_x86-64_mac_Nightly_testList_0/85

**Comment by jdekonin** [comment]
Downloaded an update to osx1013-x86-3, reboot it decided to not come back up.  I'll see if we can get some onsite aid to get it back operational.

**Comment by jdekonin** [comment]
Recovered a mac that wasn't on OpenJ9 jenkins.  Build and test for JDK11 in progress now for mac10-x86-3.  No labels so it won't cause problems with PR or nightly builds.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/178/

**Comment by jdekonin** [comment]
osx1013-x86-3 -> mac11-x86-2
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/186/
I missed making the changes for new method of supplying changes via sysctl.conf.  Respun grinders after the update to verify
https://openj9-jenkins.osuosl.org/job/Grinder/1061/
https://openj9-jenkins.osuosl.org/job/Grinder/1062/

**Comment by jdekonin** [comment]
osx1014-x86-3 -> mac12-x86-1. Upgrade appeared to go seamlessly.  Personal build to verify.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/210/

**Comment by jdekonin** [comment]
osx1014-x86-1 -> mac11-x86-3.  Upgrade appeared to go seamlessly. Personal build to verify.https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/211/

**Comment by jdekonin** [comment]
osx1014-x86-2 -> mac10-x86-5.  Upgrade appeared to go seamlessly. Personal build to verify.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/212/
